packages/javascript/bh-shared-ui/src/utils/parseItemId.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+export const parseItemId = (itemId: string) => {
+    // Edge identifiers can be either `rel_<sourceNodeId>_<edgeKind>_<targetNodeId>`...
+    let match = itemId.match(/^(?:rel_)?(\d+)_(.+)_(\d+)$/);
```

Comment:
I thiiiiink there is a possibility of this being a vulnerability because we could have user input getting dropped directly into the regex, im not sure if we need to sanitize or escape anything. Id like to spend a little time researching this. 

I do really like this approach for tho and think its very handy!!

Reply:
jk I was thinking about this wrong - my concern would be if we used the user supplied value in the regex, but not at all the case.